src/tiled/automapper.cpp

```diff
@@ -691,7 +691,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 bool matchListNo  = false;
 
 
-                if (!setLayer->contains(x + offset.x(), y + offset.y())) {
+                if (!(setLayer->contains(x + offset.x(), y + offset.y()) || setLayer->map()->infinite())) {
```

Comment:
What part did you try removing exactly? I suggested removing this entire bit of code:
```c++
                if (!(setLayer->contains(x + offset.x(), y + offset.y()) || setLayer->map()->infinite())) {
                    foreach (const TileLayer *comparedTileLayer, listYes) {
                        if (!comparedTileLayer->contains(x, y))
                            return false;

                        const Cell &c2 = comparedTileLayer->cellAt(x, y);
                        if (!c2.isEmpty())
                            return false;
                    }
                    continue;
                }
```
It's not clear to me what is happening from those screenshots.